core/creator/src/main/java/io/quarkus/creator/phase/runnerjar/RunnerJarPhase.java

```diff
@@ -346,6 +340,52 @@ public void accept(Path path) {
         }
     }
 
+    /**
+     * Manifest generation is quite simple : we just have to push some attributes in manfest.
+     * However, it gets a little more complex is manifest preexists.
+     * So we first try to see if a manifest exists, and otherwise create a new one.
+     * 
+     * <b>BEWARE</b> this method should be invoked after file copy from target/classes and so on.
+     * Otherwise this manifest manipulation will be useless.
+     * 
+     * @see https://github.com/quarkusio/quarkus/issues/1443
+     * @param runnerZipFs
+     * @param classPath
+     * @throws IOException
+     */
+    private void generateManifest(FileSystem runnerZipFs, final String classPath) throws IOException {
+        final Path manifestPath = runnerZipFs.getPath("META-INF", "MANIFEST.MF");
+        final Manifest manifest = new Manifest();
+        if (Files.exists(manifestPath)) {
+            log.warn(
+                    "Seems like you have a pre-existing MANIFEST.MF. Since quarkus generates an executable JAR, some entries will be added.");
```

Comment:
I wouldn't have a warn here. It's not something that needs special attention or something on which the user can do something.